finetune/model.py

```diff
@@ -529,14 +557,71 @@ def finetune(self, X, Y, batch_size=BATCH_SIZE, val_size=0.05, val_interval=150)
         :param val_size: Float fraction or int number that represents the size of the validation set.
         :param val_interval: The interval for which validation is performed, measured in number of steps.
         """
+        self.is_classification = True
         return self._finetune(X, Y=Y, batch_size=batch_size, val_size=val_size, val_interval=val_interval)
 
 
+class LanguageModelGeneralAPI(LanguageModelBase):  # TODO (BEN) add regression vs classification.
+    def __init__(self, autosave_path, max_length=MAX_LENGTH, verbose=True, is_classification=None):
+        super().__init__(autosave_path=autosave_path, max_length=max_length, verbose=verbose)
+        self.is_classification = is_classification
```

Comment:
I feel like this should never really be passed in at initialization.

Reply:
If it was set at initialization I would make it something more like, target_type='classification' or target_type='regression'